brainpad/courses/programming/vb/threading_and_events.md

```diff
@@ -1 +1,324 @@
 # Threading and Events
+
+## Functions
+The `BrainPad` object (or class) includes **functions** to control many aspects of the BrainPad's hardware. A function is a set of instructions grouped together. If a student is asked to speak, the command may look like `Student.Say("Hello")`. The `Say()` function is simple, but speaking requires many things like taking in air and moving your vocal cords. In the same sense, activating the LightBulb and Turning it green is a simple request but internally it does many small tasks to reach the final goal.
+Functions can also take arguments. For example, you could have a function called `Student.Run()` to order a student to run or `Student.Run(slow)` to order them to run slow. Functions can also return a value, like `Student.GetAge()` which returns the student's age.
+
+In the code example below, we show how a function we create called `Add` can be used to add 5 + 2 and print the total integer to the Output window. We've also added an additional line to display the result on the BrainPad's own display too. 
+```vb
+Class Program
+    Public Sub BrainPadSetup()
+        Dim total As Integer
+        total = Add(5, 2)
+        BrainPad.WriteToComputer(total)
+        BrainPad.Display.DrawNumberAndShowOnScreen(1, 1, total)
+    End Sub
+    Public Sub BrainPadLoop()
```

Comment:
Same with C#, functions should have a blank line before and after them, except when they're the first or last one in the given block.